theano/gof/wrapper.py

```diff
@@ -0,0 +1,443 @@
+    # you should import required Theano types.
+    from theano.tensor import TensorType
+    from theano.scalar import Scalar
+
+In an op you create:
+
+.. code-block:: python
+
+    params_type = Wrapper(attr1=TensorType('int32', (False, False)), attr2=Scalar('float64'))
+
+If your op contains attributes ``attr1`` **and** ``attr2``, the default ``op.get_params()``
+implementation will automatically try to look for it and generate an appropriate Wrap object.
+Attributes must be compatible with the corresponding types defined into the Wrapper
+(we will try to convert and downcast if needed). For example, ``your_op.attr1``
+should be a matrix of integers, and ``your_op.attr2``
+should be a real number (integer or floating value).
+
+.. code-block:: python
+
+    def __init__(value_attr1, value_attr2):
+        self.attr1 = value_attr1
+        self.attr2 = value_attr2
+
+In ``perform()`` implementation (with params named ``param``):
+
+.. code-block:: python
+
+    matrix_param = param.attr1
+    number_param = param.attr2
+
+In ``c_code()`` implementation (with ``param = sub['params']``):
+
+.. code-block:: c
+
+    PyArrayObject* matrix = param->attr1;
+    npy_float64    number = param->attr2;
+    /* You won't need to free them or whatever else. */
+
+
+See :class:`QuadraticOpFunc` and :class:`QuadraticCOpFunc` in ``theano/gof/tests/test_wrapper.py``
+for complete working examples.
+
+"""
+
+from __future__ import absolute_import, print_function, division
+import re
+import hashlib
+from theano.gof.utils import MethodNotDefined, c_cpp_keywords
+from theano.gof import Type
+
+
+class Wrap(dict):
+    """
+    Internal convenient class to wrap many Python objects into one
+    (this class is not safe as the hash method does not check if values are effectively hashable).
+
+    **Example:**
+
+    .. code-block:: python
+
+        from theano.gof import Wrapper, Wrap
+        from theano.scalar import Scalar
+        # You must create a Wrapper first:
+        wp = Wrapper(attr1=Scalar('int32'), key2=Scalar('float32'), field3=Scalar('int64'))
+        # Then you can create a Wrap with the wrapper defined above and values for attributes.
+        w = Wrap(wp, attr1=1, key2=2.0, field3=3)
+        print(w.attr1, w.key2, w.field3)
+        d = dict(attr1=1, key2=2, field3=-1)
+        w2 = Wrap(wp, **d)
+        print(w2.attr1, w2.key2, w2.field3)
+
+    """
+
+    def __init__(self, wrapper, **kwargs):
+        if not isinstance(wrapper, Wrapper):
+            raise TypeError('Wrap: 1st constructor argument should be a Wrapper.')
+        for field in wrapper.fields:
+            if field not in kwargs:
+                raise TypeError('Wrap: Wrapper attribute "%s" not in Wrap args.' % field)
+        super(Wrap, self).__init__(**kwargs)
+        self.__dict__.update(wrapper=wrapper)
+
+    def __repr__(self):
+        return 'Wrap(%s)' % ', '.join([('%s:%s' % (k, type(self[k]))) for k in sorted(self.keys())])
```

Comment:
Ah ok ! I wanted to avoid a too long printing when a Wrap contain big numpy ndarrays, for example. Also, I would have to print the associated Wrapper, too.

Can I use `__str__()` instead, and just remove `__repr__()` implementation ?